tests/conftest.py

```diff
@@ -77,9 +77,18 @@ def compare_df(
     for col in floats:
         actual_col = actual.loc[expected.index, col].values
         expected_col = expected[col].values
-        if actual_col != approx(expected_col, rel=rtol, abs=atol, nan_ok=equal_nan):
-            print(f"file: {msg}, column: {col}")
-        assert actual_col == approx(expected_col, rel=rtol, abs=atol, nan_ok=equal_nan)
+        try:
+            assert actual_col == approx(
+                expected_col, rel=rtol, abs=atol, nan_ok=equal_nan
+            )
+        except AssertionError:
+            # if the columns are not equal, we check if the sets are equal as sometimes
+            # the order of the rows is different because of different sorting algorithms
+            try:
+                assert set(actual_col) == set(expected_col)
+            except AssertionError:
+                print(f"file: {msg}, column: {col}")
+                raise
```

Comment:
Was this debug printing that got left behind?

```suggestion
            assert set(actual_col) == set(expected_col)
```

Besides that, this seems a bit hacky. (Also, given that you're not using `approx()` for the second attempt at comparing, I worry it's maybe a bit fragile.) I'm not against it if it's the only sensible way to do this, but I wonder if we might be able to do something cleaner... Is there no way to sort the rows for `actual_col` and `expected_col` in the same way here, then do the comparison?

Reply:
Yes, I was thinking the same thing after implementing this. What do you think of the following?

```python
assert np.sort(actual_col) == approx(
                np.sort(expected_col), rel=rtol, abs=atol, nan_ok=equal_nan
            )
```

Reply:
That looks better :smile: